src/axom/sina/core/Document.cpp

```diff
@@ -370,5 +376,771 @@ Document loadDocument(std::string const &path, RecordLoader const &recordLoader,
+          std::vector<std::string> curve_sets_list;
+          existing_file.list_child_names("records/" + rec_name + "/curve_sets", curve_sets_list);
+          auto &new_curve_sets = record->getCurveSets();
+
+          for(const auto &curve_set_pair : new_curve_sets)
+          {
+            const std::string &cs_name = curve_set_pair.first;
+            const auto &cs_data = curve_set_pair.second;
+
+            bool cs_exists = (std::find(curve_sets_list.begin(), curve_sets_list.end(), cs_name) !=
+                              curve_sets_list.end());
+
+            if(cs_exists)
+            {
+              const auto &new_dependents = cs_data.getDependentCurves();
+              for(const auto &dep_pair : new_dependents)
+              {
+                const std::string &dep_name = dep_pair.first;
+                std::string curve_set_path =
+                  record_path + "/curve_sets/" + cs_name + "/dependent/" + dep_name + "/value";
+
+                if(info.has_path(curve_set_path) && info[curve_set_path].has_child("num_elements"))
+                {
+                  int current_size = info[curve_set_path]["num_elements"].to_int();
+                  opts["offset"] = current_size;
+                }
+                else
+                {
+                  opts["offset"] = 0;
+                }
+
+                const auto &new_dep_values = dep_pair.second.getValues();
+                std::vector<double> current_array(new_dep_values.begin(), new_dep_values.end());
+                to_set.set(current_array);
+
+                write_queue.push_back([curve_set_path, hdf5FilePath, opts, to_set]() mutable {
+                  conduit::relay::io::hdf5_write(to_set, hdf5FilePath, curve_set_path, opts, true);
+                });
+              }
+
+              const auto &new_independents = cs_data.getIndependentCurves();
+              for(const auto &indep_pair : new_independents)
+              {
+                const std::string &indep_name = indep_pair.first;
+                std::string curve_set_path =
+                  record_path + "/curve_sets/" + cs_name + "/independent/" + indep_name + "/value";
+
+                if(info.has_path(curve_set_path) && info[curve_set_path].has_child("num_elements"))
+                {
+                  int current_size = info[curve_set_path]["num_elements"].to_int();
+                  opts["offset"] = current_size;
+                }
+                else
+                {
+                  opts["offset"] = 0;
+                }
+
+                const auto &new_indep_values = indep_pair.second.getValues();
+                std::vector<double> current_array(new_indep_values.begin(), new_indep_values.end());
+                to_set.set(current_array);
+
+                write_queue.push_back([curve_set_path, hdf5FilePath, opts, to_set]() mutable {
+                  conduit::relay::io::hdf5_write(to_set, hdf5FilePath, curve_set_path, opts, true);
```

Comment:
In the same vein as my comment above -- does it makes sense to have so many `write` calls (one per curve), rather than one call at the end that writes out the entire file.

Perhaps I'm missing something and appends are actually required by the application vs. rewriting the file.